### ISSUE TITLE
Duplicated error definitions in the ABI

### ISSUE THREAD

**Issue by ultrasecreth** [ethereum/solidity#14404]
## Description

If the same error is defined in different files within the inheritance/use chain, it ends up twice in the ABI despite having the same selector. 
The generated bytecode seems fine as I couldn't find the selector being duplicated in it.

## Environment

- Compiler version: 0.8.20
- Target EVM version (as per compiler settings): London
- Framework/IDE (e.g. Truffle or Remix): Foundry
- EVM execution environment / backend / blockchain client: N/A
- Operating system: MacOS

## Steps to Reproduce

```solidity
pragma solidity 0.8.20;

library A {
    error CustomError();

    function iCustomError() internal pure {
        revert CustomError();
    }
}

library B {
    error CustomError();

    function iCustomError() internal pure {
        revert CustomError();
    }
}

contract SolidityBug {
    function foo() public pure {
        A.iCustomError();
    }

    function bar() public pure {
        B.iCustomError();
    }
}
```

will generate 

```json
  "abi": [
    {
      "inputs": [],
      "name": "CustomError",
      "type": "error"
    },
    {
      "inputs": [],
      "name": "CustomError",
      "type": "error"
    },
    {
      "inputs": [],
      "name": "bar",
      "outputs": [],
      "stateMutability": "pure",
      "type": "function"
    },
    {
      "inputs": [],
      "name": "foo",
      "outputs": [],
      "stateMutability": "pure",
      "type": "function"
    }
  ]
```

I know this seems silly, but the real world example on how this can happen is the following. Here the `AddressEmptyCode` error will be defined in both OZ libs, and it'll end up twice in the ABI. (need to use the latest master from OZ)

```solidity
import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";

contract SolidityBug is UUPSUpgradeable
{
    function _authorizeUpgrade(address newImplementation) internal override {}

    function useSafeERC20(IERC20 token, address to, uint256 amount) external {
        SafeERC20.safeTransferFrom(token, msg.sender, to, amount);
    }
}
```

**Comment by agnxsh** [comment]
I would like to take up this issue :D

**Comment by cameel** [comment]
I don't think this is a bug. It's been explicitly documented that way [in the ABI JSON description for errors and events](https://docs.soliditylang.org/en/latest/abi-spec.html#json) ever since errors were introduced in #10937.

We could still change it, but the question is what behavior is preferable in practice. Note that this happens only in situations where in the code there are actually two distinct definitions that happen to have the same signature. Is this causing actual issues?

I remember that a similar thing was discussed about events, back when we changed the ABI JSON to include all emitted ones: https://github.com/ethereum/solidity/pull/10996#discussion_r590515665. Some filtering of identical events was proposed, but we ended up not doing that. If we change it for errors, we should change it for events as well.
